sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java

```diff
@@ -52,25 +48,31 @@ public ChooseSourceCaseSteps(
         "^I open the first found result in the CHOOSE SOURCE window$",
         () -> {
           webDriverHelpers.clickOnWebElementBySelector(SOURCE_CASE_WINDOW_FIRST_RESULT_OPTION);
-          webDriverHelpers.clickWhileOtherButtonIsDisplayed(
-              SOURCE_CASE_WINDOW_CONFIRM_BUTTON, CHANGE_CASE_BUTTON);
+          webDriverHelpers.waitUntilElementIsVisibleAndClickable(SOURCE_CASE_WINDOW_CONFIRM_BUTTON);
+          webDriverHelpers.clickOnWebElementBySelector(SOURCE_CASE_WINDOW_CONFIRM_BUTTON);
+          webDriverHelpers.waitUntilIdentifiedElementIsVisibleAndClickable(
+              SUCCESSFUL_CASE_CHANGE_POPUP);
         });
 
     When(
         "I click on the CHOOSE SOURCE CASE button",
```

Comment:
I know that is out of this PR, but can we change the name of the gherkin syntax to reflect this action?

Reply:
reworked